### PR TITLE
Add documentation with Documenter.jl

### PR DESCRIPTION
**Pull request by hyrodium**
This PR adds documentation with Documenter.jl. Note that the documentation is still incomplete.

image

### CODE REVIEW

File: docs/Project.toml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[deps]
+Documenter = "e30172f5-a6a5-5a46-863b-614d45cd2de4"
+Quaternions = "94ee1d12-ae83-5a48-8b1c-48b8ff168ae0"
```
Comment (sethaxen) [inline docs/Project.toml:3]:
```suggestion
```
It's not necessary to include the UUID of the package itself for its docs

File: docs/src/index.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# Quaternions.jl
+
+A Julia module with [quaternion](https://en.wikipedia.org/wiki/Quaternion), [octonion](https://en.wikipedia.org/wiki/Octonion) and [dual-quaternion](https://en.wikipedia.org/wiki/Dual_quaternion) functionality
```
Comment (sethaxen) [inline docs/src/index.md:3]:
```suggestion
A Julia package implementing [quaternions](https://en.wikipedia.org/wiki/Quaternion), [octonions](https://en.wikipedia.org/wiki/Octonion) and [dual-quaternions](https://en.wikipedia.org/wiki/Dual_quaternion)
```

**Review by sethaxen — COMMENTED** [review]
Thanks for setting this up! Just a few minor suggestions.

**Comment by hyrodium** [comment]
The documentation is now deployed! :tada:
https://juliageometry.github.io/Quaternions.jl/dev/

**Pull request merged**